packages/contracts/test/contracts/chugsplash/L1ChugSplashProxy.spec.ts

```diff
@@ -0,0 +1,180 @@
+      await signer1.getAddress()
+    )
+  })
+
+  describe('getOwner', () => {
+    it('should return the owner if called by the owner', async () => {
+      expect(
+        await L1ChugSplashProxy.connect(signer1).callStatic.getOwner()
+      ).to.equal(await signer1.getAddress())
+    })
+
+    it('should return the owner if called by the zero address in an eth_call', async () => {
+      expect(
+        await L1ChugSplashProxy.connect(signer1.provider).callStatic.getOwner({
+          from: hre.ethers.constants.AddressZero,
+        })
+      ).to.equal(await signer1.getAddress())
+    })
+
+    it('should otherwise pass through to the proxied contract', async () => {
+      await expect(
+        L1ChugSplashProxy.connect(signer2).callStatic.getOwner()
+      ).to.be.revertedWith('L1ChugSplashProxy: implementation is not set yet')
+    })
+  })
+
+  describe('setOwner', () => {
+    it('should succeed if called by the owner', async () => {
+      await expect(
+        L1ChugSplashProxy.connect(signer1).setOwner(await signer2.getAddress())
+      ).to.not.be.reverted
+
+      expect(
+        await L1ChugSplashProxy.connect(signer2).callStatic.getOwner()
+      ).to.equal(await signer2.getAddress())
+    })
+
+    it('should otherwise pass through to the proxied contract', async () => {
+      await expect(
+        L1ChugSplashProxy.connect(signer2).setOwner(await signer1.getAddress())
+      ).to.be.revertedWith('L1ChugSplashProxy: implementation is not set yet')
+    })
+  })
+
+  describe('getImplementation', () => {
+    it('should succeed if called by the owner', async () => {
+      expect(
+        await L1ChugSplashProxy.connect(signer1).callStatic.getImplementation()
+      ).to.equal(hre.ethers.constants.AddressZero)
+    })
+
+    it('should otherwise pass through to the proxied contract', async () => {
+      await expect(
+        L1ChugSplashProxy.connect(signer2).getImplementation()
+      ).to.be.revertedWith('L1ChugSplashProxy: implementation is not set yet')
+    })
+  })
+
+  describe('setStorage', () => {
+    it('should succeed if called by the owner', async () => {
+      const storageKey = hre.ethers.utils.keccak256('0x1234')
+      const storageValue = hre.ethers.utils.keccak256('0x5678')
+
+      await expect(
+        L1ChugSplashProxy.connect(signer1).setStorage(storageKey, storageValue)
+      ).to.not.be.reverted
+
+      expect(
+        await hre.ethers.provider.getStorageAt(
+          L1ChugSplashProxy.address,
+          storageKey
+        )
+      ).to.equal(storageValue)
+    })
+
+    it('should otherwise pass through to the proxied contract', async () => {
+      const storageKey = hre.ethers.utils.keccak256('0x1234')
+      const storageValue = hre.ethers.utils.keccak256('0x5678')
+
+      await expect(
+        L1ChugSplashProxy.connect(signer2).setStorage(storageKey, storageValue)
+      ).to.be.revertedWith('L1ChugSplashProxy: implementation is not set yet')
```

Comment:
A slightly better test (here and in the other 'pass through' tests) might be to ensure that `setCode` the implementation is called with `bytes4(keccak256("setStorage(bytes32)"))`.